pymc3/distributions/continuous.py

```diff
@@ -478,87 +462,67 @@ class Normal(Continuous):
             x = pm.Normal('x', mu=0, tau=1/23)
     """
 
-    def __init__(self, mu=0, sigma=None, tau=None, sd=None, **kwargs):
+    @classmethod
+    def dist(cls, mu=0, sigma=None, tau=None, sd=None, **kwargs):
         if sd is not None:
             sigma = sd
         tau, sigma = get_tau_sigma(tau=tau, sigma=sigma)
-        self.sigma = self.sd = tt.as_tensor_variable(sigma)
-        self.tau = tt.as_tensor_variable(tau)
+        sigma = tt.as_tensor_variable(sigma)
 
-        self.mean = self.median = self.mode = self.mu = mu = tt.as_tensor_variable(floatX(mu))
-        self.variance = 1.0 / self.tau
+        # sd = sigma
+        # tau = tt.as_tensor_variable(tau)
+        # mean = median = mode = mu = tt.as_tensor_variable(floatX(mu))
+        # variance = 1.0 / self.tau
 
         assert_negative_support(sigma, "sigma", "Normal")
-        assert_negative_support(tau, "tau", "Normal")
+        # assert_negative_support(tau, "tau", "Normal")
 
-        super().__init__(**kwargs)
+        rv_var = normal(mu, sigma, **kwargs)
+        rv_var.tag.transform = kwargs.get("transform", cls.default_transform)
 
-    def random(self, point=None, size=None):
-        """
-        Draw random values from Normal distribution.
+        return rv_var
 
-        Parameters
-        ----------
-        point: dict, optional
-            Dict of variable values on which random values are to be
-            conditioned (uses default point if not specified).
-        size: int, optional
-            Desired size of random sample (returns one sample if not
-            specified).
 
-        Returns
-        -------
-        array
-        """
-        mu, tau, _ = draw_values([self.mu, self.tau, self.sigma], point=point, size=size)
-        return generate_samples(
-            stats.norm.rvs, loc=mu, scale=tau ** -0.5, dist_shape=self.shape, size=size
-        )
-
-    def logp(self, value):
-        """
-        Calculate log-probability of Normal distribution at specified value.
+@_logp.register(NormalRV)
+def normal_logp(op, value, mu, sigma):
+    """
+    Calculate log-probability of Normal distribution at specified value.
 
-        Parameters
-        ----------
-        value: numeric
-            Value(s) for which log-probability is calculated. If the log probabilities for multiple
-            values are desired the values must be provided in a numpy array or theano tensor
+    Parameters
+    ----------
+    value: numeric
+        Value(s) for which log-probability is calculated. If the log probabilities for multiple
+        values are desired the values must be provided in a numpy array or theano tensor
 
-        Returns
-        -------
-        TensorVariable
-        """
-        sigma = self.sigma
-        tau = self.tau
-        mu = self.mu
+    Returns
+    -------
+    TensorVariable
+    """
+    tau, sigma = get_tau_sigma(tau=None, sigma=sigma)
```

Comment:
Will this be any more inefficient here than how we were doing before (calling it once during initialization)?

Reply:
Or more generally, do you foresee any complications with using alternative parametrizations (e.g., tau/ sigma)?

Reply:
We can do all the same initialization that we're doing now.  If it's not happening here, then it's only out of convenience.